numpy/core/arrayprint.py

```diff
@@ -1175,6 +1175,28 @@ def dtype_is_implied(dtype):
     return dtype.type in _typelessdata
 
 
+def dtype_short_repr(dtype):
```

Comment:
This feels like it might belong at `dtype.__format__` in future.

Reply:
This code is also duplicate from arrayprint.py, I was just modifying it in #10032. Maybe I can rebase than on this PR and we can move the function.

Reply:
I'm confused - this file _is_ `arrayprint` - where is the other duplicate?

Reply:
Do you want me to split off a PR with just the first two commits, which we can merge first?

Reply:
Oh sorry I thought this was `ma/core.py`.  This change is fine.